src/main/java/pl/sda/library/infrastructure/jpa/AuthorRepositoryAdapter.java

```diff
@@ -16,36 +19,46 @@
 
     @Override
     public Long addAuthor(Author author) {
-        return null;
+        if(isExistByFirstNameAndLastName(author.getFirstName(), author.getLastName())){
+            throw new AuthorAlreadyExistException();
+        }
+        return authorDtoRepository.save(AuthorDto.fromDomain(author)).getId();
     }
 
     @Override
     public Optional<Author> getAuthorById(Long id) {
+        Optional<AuthorDto> byId = authorDtoRepository.findById(id);
+        if(byId.isPresent()){
+            return Optional.of(byId.get().toModel());
+        }
         return Optional.empty();
     }
 
     @Override
     public List<Author> getAllAuthors() {
-        return null;
+        List<Author> authorList = new ArrayList<>();
+        authorDtoRepository.findAll().iterator().forEachRemaining(authorDto -> authorList.add(authorDto.toModel()));
+        return authorList;
     }
 
     @Override
     public void deleteAuthor(long id) {
-
+        if(!isExistById(id)){
```

Comment:
w nomenklaturze restowej metoda DELETE jest idempotentna
to znaczy, ze kolejne wywolanie delete na ten sam zasob powinno zwrocic 200
nie ma potrzeby sprawdzania czy author istnieje

Reply:
plus metoda moze sie nazywac po prostu delete

Reply:
ok

Reply:
jeżeli usuwam autora którego nie ma to error 500

Reply:
to w takim razie w tym if róbmy return, a nie rzucajmy wyjatkiem

Reply:
Możesz w sumie zrobic tez inaczej
niech ta metoda zwraca boolean

jesli uda się usunać to zwracany jest true, jezeli nie to false

wtedy w service bedziesz mogl sobie okodować zachowanie

Reply:
dodać try catch w adapterze?